### DISCUSSION TITLE
Unable to update KEDA / delete scaledobjects.keda.sh

### DISCUSSION THREAD

**Discussion by Vincentvwal** [kedacore/keda#2159] [Q&A / Need Help] [answered]
We were running KEDA 2.1.1 on our AKS cluster and we are currently in the process of including this in our update process.
Initially we tried running helm upgrade on our installation to update to the latest version, which failed as there were files already available.

So we ran helm uninstall which uninstalled KEDA but left the CustomResourceDefinitions intact.
Removing clustertriggerauthentications.keda.sh and scaledjobs.keda.sh was no problem, however removing scaledobjects.keda.sh just wont work.

We get the message **customresourcedefinition.apiextensions.k8s.io "scaledobjects.keda.sh" deleted** however the file remains.
To top this off, we cannot remove the ScaledObject as well anymore, installing KEDA again greets us with the error **Error: rendered manifests contain a resource that already exists. Unable to continue with install: existing resource conflict: kind: CustomResourceDefinition, namespace: , name: scaledobjects.keda.sh**

This means we are kind of stuck on this.
Luckily this our acceptance enviroment which hardly ever scaled, but still we would like to get it back to work.

It seems like a finalizer is still on it, however i am unable to remove this using `kubectl patch crd scaledobjects.keda.sh -p '{"metadata":{"finalizers":[]}}' --type=merge` as it tells me it is invalid JSON Patch

**Comment by zroubalik** [comment, marked as answer]
There were some problems with updating older versions of KEDA and CRDs. @tomkerkhove do you have any idea how to tackle this?

**Reply by Vincentvwal** [reply]
Sorry, managed to tackle this issue this morning.
My patch was incorrect as the quotes needed escaping

`kubectl patch crd scaledobjects.keda.sh -p '{\"metadata\":{\"finalizers\":[]}}' --type=merge`

As well as running `kubectl delete -f https://raw.githubusercontent.com/kedacore/keda/main/config/crd/bases/keda.sh_scaledobjects.yaml` the scaledobjects.keda.sh was removed and we could re-install using helm.

I have added a full uninstall procedure on our end which currently is

helm uninstall -n keda keda
kubectl delete -f https://raw.githubusercontent.com/kedacore/keda/main/config/crd/bases/keda.sh_scaledobjects.yaml
kubectl delete -f https://raw.githubusercontent.com/kedacore/keda/main/config/crd/bases/keda.sh_scaledjobs.yaml
kubectl delete -f https://raw.githubusercontent.com/kedacore/keda/main/config/crd/bases/keda.sh_triggerauthentications.yaml
kubectl delete -f https://raw.githubusercontent.com/kedacore/keda/main/config/crd/bases/keda.sh_clustertriggerauthentications.yaml

**Reply by zroubalik** [reply]
Awesome glad to hear that, please be sure that you have up to date version (2.4) :)

**Reply by Vincentvwal** [reply]
Will do and have it!

Thanks for the help!
